perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -5538,6 +5538,65 @@ public void addGroupsToAutoRegistration(PerunSession sess, List<Group> groups) {
 		}
 	}
 
+	@Override
+	public List<List<Group>> getIndirectMembershipPaths(PerunSession sess, Member member, Group group) throws MemberNotExistsException, GroupNotExistsException {
+		if (!isGroupMember(sess, group, member))
+			throw new MemberNotExistsException("Member does not belong to this group");
+
+		Group sourceGroup = getGroupById(sess, member.getSourceGroupId());
```

Comment:
This would typically throw NullPointerException because sourceGroupId is null when this method is called from outer API. SourceGroupId is actually set only in a member context to a group (e.g. when getting group members sourceGroupId is set in context to this group), but the member is determined in outer API only by ID, not by any groups. It works in tests only because the member object has this property set when the member is added to groups in tests.
Also, one member can have multiple source groups = groups where the member is direct.

Reply:
Good point, now it should be simplified to just check if member is direct member.